pin/views.py

```diff
@@ -1,4 +1,300 @@
+        return JsonResponse({"pins" : pins_detail})
+
+class CommentView(View):
+    @login_required
+    def post(self, request, **kwargs):
+        user                    = kwargs['user']
+        user_id                 = kwargs['user_id']
+        pin_id                  = kwargs['pin']
+        body                    = json.loads(request.body)
+        comment                 = body['comment']
+        pin                     = Pin.objects.get(id=pin_id)
+        
+        new_comment             = Comment.objects.create(content=comment, pin=pin, user=user)
+        pin_comment             = pin.comments.all()
+        pin_comment_info = [{
+            "user": comment.user.email.split("@")[0] if comment.user.email else comment.user.social_id,
+            "comment": model_to_dict(comment)
+        } for comment in pin_comment]
+        comment_len             = len(pin_comment_info)
+        pin_user_data           = serializers.serialize('json', pin.user.all())
+
+        return JsonResponse(
+            {"pin": {
+                "id": pin.id, 
+                "image_url": pin.image_url, 
+                "text1": pin.paragraph1, 
+                "text2": pin.paragraph2, 
+                "users": pin_user_data
+                }, 
+            "comment"       : pin_comment_info, 
+            "comment_total" : comment_len
+            })
+
+    @login_required
+    def delete(self, request, **kwargs):
+        user        = kwargs['user']
+        user_id     = kwargs['user_id']
+        pin_id      = kwargs['pin']
+        body        = json.loads(request.body)
+        comment_id  = body['comment_id']
+        pin         = Pin.objects.get(id=pin_id)
+    
+        Comment.objects.get(id=comment_id).delete()
+        pin_comment             = pin.comments.all()
+        pin_comment_info = [{
+            "user": comment.user.email.split("@")[0] if comment.user.email else comment.user.social_id,
+            "comment": model_to_dict(comment)
+        } for comment in pin_comment]
+        
+        comment_len             = len(pin_comment_info)
+        pin_user_data           = serializers.serialize('json', pin.user.all())
+
+        return JsonResponse({
+            "pin": {
+                "id": pin.id, 
+                "image_url" : pin.image_url, 
+                "text1"     : pin.paragraph1, 
+                "text2"     : pin.paragraph2, 
+                "users"     : pin_user_data
+                }, 
+                "comment"       : pin_comment_info, 
+                "comment_total" : comment_len
+            })
+
+    @login_required
+    def get(self, request, **kwargs):
+        user                    = kwargs['user']
+        user_id                 = kwargs['user_id']
+    
+        pin_id                  = kwargs['pin']
+        pin                     = Pin.objects.get(id=pin_id)
+        pin_comment             = pin.comments.all()
+        pin_comment_info        = []
+        pin_comment_info = [{
```

Comment:
pin_comment_info 는 리스트 컴프리헨션으로 리스트를 만들어주고있기 때문에 빈리스트를 위에서 선언하지않아도 됩니다.
```suggestion
        
        pin_comment_info = [{
```